security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/DefaultSecureEndpoint.java

```diff
@@ -48,7 +50,28 @@ public String getUrl() {
     }
 
     @Override
-    public Optional<List<AuthenticationMethod>> getSupportedAuthenticationMethods() {
+    public Optional<List<String>> getAuthenticationMethodsSupported() {
```

Comment:
It might be better to have `Set` if the collection has unique items and `contains` is used

Reply:
Does it need to be optional?

Reply:
I [changed to `Set<String>`](https://github.com/micronaut-projects/micronaut-security/pull/1790/commits/5685b551d623b33e86974e928fa69ae4fe35f40c) and I removed the [optional from the return type](https://github.com/micronaut-projects/micronaut-security/pull/1790/commits/574ec725738ee427b855a3c685702d4e01314ab4).